src/rust/src/main.rs

```diff
@@ -0,0 +1,32 @@
+extern crate rand;
+
+use std::vec::Vec;
+use rand::distributions::{IndependentSample, Range};
+
+fn main() {
+    let mut lotto_numbers = Vec::new();
+    for n in 0..45 {
+        lotto_numbers.push(n+1);
+    }
+
+    let mut rng = rand::thread_rng();
+    let mut result_numbers = Vec::new();
+    for _ in 0..6 {
+       	let between = Range::new(0, lotto_numbers.len()-1);
+
+        let index = between.ind_sample(&mut rng);
+        result_numbers.push(lotto_numbers[index]);
+        lotto_numbers.remove(index);
+    }
```

Comment:
I think you could try to use `rand::sample` rather than implement.

https://doc.rust-lang.org/rand/rand/fn.sample.html